src/main/java/com/denizenscript/depenizen/bukkit/events/chestshop/ChestShopTransactionScriptEvent.java

```diff
@@ -0,0 +1,78 @@
+package com.denizenscript.depenizen.bukkit.events.chestshop;
+
+import com.Acrobot.ChestShop.Events.TransactionEvent;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class ChestShopTransactionScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // chestshop transaction
+    //
+    // @Triggers when a transaction occurs at a chestshop.
+    //
+    // @Switch type:<type> to only process if the transaction was either a "buy" or "sell" event.
+    //
+    // @Cancellable true
+    //
+    // @Context
+    // <context.container> returns a LocationTag of the container attached to the shop, if any.
+    // <context.item> returns an ItemTag of the item involved in the transaction.
+    // <context.money> returns an ElementTag(Decimal) of the amount of money involved in the transaction.
+    // <context.sign> returns a LocationTag of the sign running the shop.
+    // <context.type> returns whether the transaction was a "BUY" or "SELL" event.
```

Comment:
The word `whether` implies a boolean (true/false).

`<context.type> returns the type of transaction that occurred, either "BUY" or "SELL".`